### PR TITLE
Add istio-init to manage istio CRDs

### PR DESCRIPTION
**Pull request by guangbochen**
**Problem:**

1. Re-enable istio app using native Helm will cause CRD already exists https://github.com/rancher/rancher/issues/20843
2. Istio CRDs are not removed after disabling Istio https://github.com/rancher/rancher/issues/20831

**Solution:**
Add `isito-init` sub-chart to handle istio CRDs installation, upgrade and remove. 

**Related Issues:**
https://github.com/rancher/rancher/issues/20843
https://github.com/rancher/rancher/issues/20831

### CODE REVIEW

**Comment by loganhz** [comment]
LGTM

File: charts/rancher-istio/0.0.2/charts/istio-init/templates/job-crd-10.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+spec:
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "false"
+    spec:
+      serviceAccountName: istio-init-service-account
+      containers:
+      - name: istio-init-crd-10
+        image: "{{ template "system_default_registry" . }}{{ .Values.global.kubectl.repository }}:{{ .Values.global.kubectl.tag }}"
```
Comment (gitlawr) [inline charts/rancher-istio/0.0.2/charts/istio-init/templates/job-crd-10.yaml:15]: Is there any default value for `global.kubectl.xxx` in the values file?

File: charts/rancher-istio/0.0.2/charts/istio-init/files/crd-certmanager-10.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: clusterissuers.certmanager.k8s.io
```
Comment (gitlawr) [inline charts/rancher-istio/0.0.2/charts/istio-init/files/crd-certmanager-10.yaml:4]: I can see CRDs in both `charts/rancher-istio/0.0.2/charts/istio-init/files/crd-certmanager-10.yaml ` and `charts/rancher-istio/0.0.2/templates/crd-certmanager.yaml`. Do we need this replication? The same to Istio CRDs.

**Comment by gitlawr** [comment]
It would be easier to review if the original content of `istio-init` chart and the changes we made are separated.

File: charts/rancher-istio/0.0.2/charts/istio-init/templates/job-crd-10.yaml (hunk 1 above)
Comment (guangbochen) [reply charts/rancher-istio/0.0.2/charts/istio-init/templates/job-crd-10.yaml:15]: yes.

File: charts/rancher-istio/0.0.2/charts/istio-init/files/crd-certmanager-10.yaml (hunk 2 above)
Comment (guangbochen) [reply charts/rancher-istio/0.0.2/charts/istio-init/files/crd-certmanager-10.yaml:4]: we will need both, the crd-certmanager.yaml will pre-install CRDs on bootstrap if the crd does not exist, the `istio-init` one is responsible for future apply or update.

**Comment by guangbochen** [comment]
@gitlawr done, also for the helm unittest you may need to refer to this commit https://github.com/guangbochen/system-charts/commit/9942ab9dccef2246f5a820d492647d121bd11052, it contains all the unittest of subcharts, thanks.

**Comment by gitlawr** [comment]
LGTM

**Comment by cjellick** [comment]
lgtm

**Pull request merged**